src/test/java/org/cactoos/io/TempFolderTest.java

```diff
@@ -61,4 +67,51 @@ public void deletesDirectory() throws Exception {
             new IsTrue()
         ).affirm();
     }
+
+    @Test
+    public void deletesNonEmptyDirectory() throws Exception {
+        final TempFolder temp = new TempFolder();
+        final Path root = temp.value();
+        new ForEach<>(
+            new ProcOf<String>(
+                name -> {
+                    final Path dir = Files.createDirectories(
+                        new File(root.toFile(), name).toPath()
+                    );
+                    new ForEach<>(
+                        new ProcOf<String>(
+                            filename -> {
+                                new TempFile(
+                                    () -> dir,
+                                    filename,
+                                    ""
+                                ).value();
+                                return true;
+                            }
+                        )
+                    ).exec(
+                        new IterableOf<>(
+                            "file1.txt", "file2.txt", "file3.txt"
+                        )
+                    );
+                    return true;
+                }
+            )
+        ).exec(
+            new IterableOf<>(
+                "a", "b", "c", "d", "e"
+            )
+        );
+        new Assertion<>(
```

Comment:
@fanifieiev Let's split this into two different test methods, one for each assertion